### PR TITLE
Feat: Added question and result page

### PR DESCRIPTION
**Pull request by sushil-mod**
# Added question and result page with its functionality 

### What to check 
- Rules page
- Result page
- Quizslice

### CODE REVIEW

File: src/pages/result/Result.css [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    text-underline-offset:8px;
+}
+.result-paper-container{
+    display:flex;
+    justify-content: center;
+    width: 100%;
+}
+.result-paper{
+    font-size:1.5rem;
+    padding:10px;
```
Comment (MayankKumar10) [inline src/pages/result/Result.css:14]: use rem instead of px here

File: src/pages/result/Result.css [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    justify-content:center;
+    gap:1rem;
+    margin:0rem auto;
+}
+.result-paper-que{
+    font-size:1.5rem;
+    padding:0.5rem;
+    text-align:center;
+}
+.result-paper-question{
```
Comment (MayankKumar10) [inline src/pages/result/Result.css:31]: make a flex helper class

**Review by MayankKumar10 — APPROVED** [review]
Look into these Suggestions everything else looks good to me you can merge it.

File: src/App.tsx [hunk 3]
```diff
@@ -13,6 +14,7 @@ function App() {
       <Routes>
         <Route path='/' element={<Home/>} />
         <Route path='/quiz/:quizId' element={<Rules/>} />
+        <Route path='/result' element={<Result/>} />
       </Routes>
     </>
   )
```
Comment (therajatg) [inline src/App.tsx:20]: You can have different file for Route.

File: src/App.tsx [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { Route, Routes } from 'react-router-dom';
 import './App.css';
 import NavBar from './components/navbar/NavBar';
```
Comment (parulgupta2001) [inline src/App.tsx:4]: import all components from one file like index.js

File: src/components/card/CategoryCard.tsx [hunk 5]
```diff
@@ -10,15 +10,14 @@ function CategoryCard(Props : CategoryType) {
...
-
   return (
     <Card className='category-card-container' >
         <img className='category-card-img' src={image.srcURL} alt={image.altText} />
         <div className='text-overlay-container' >
             <div className='text-overlay' >
                 <Typography variant='h4' className='category-title'>{ categoryName }</Typography>
                 <CardActions className='overlay-btn' >
-                <Button onClick={ ()=>navigate(`/quiz/${id}`) }  className='play-btn' variant='contained' size='medium' color="primary"> Play Quiz </Button>
+                <Button onClick={ ()=>navigate(`/quiz/${id}`,{ replace: true }) }  className='play-btn' variant='contained' size='medium' color="primary"> Play Quiz </Button>
```
Comment (therajatg) [inline src/components/card/CategoryCard.tsx:20]: I like that you have this dynamic link.

**Review by therajatg — APPROVED** [review]
everything else looks good.

File: src/pages/result/Result.css [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    text-align:center;
+}
+
+.result-container{
+    display:flex;
+    flex-direction:column;
+    width:40%;
+    justify-content:center;
+    gap:1rem;
+    margin:0rem auto;
```
Comment (parulgupta2001) [inline src/pages/result/Result.css:24]: no need of any unit after zero value

**Review by parulgupta2001 — APPROVED** [review]
some changes are suggested, otherwise everything looks good.

**Review by krishanu2911 — APPROVED** [review]
Well Written Modular code No changes required
Just compile all the import in one Index file

**Comment by varun-Tad** [comment]
You can have all the imports in one file otherwise the code looks good to merge

**Pull request merged**